### PR TITLE
passive staking 4

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 passive staking is only a proposal

 #### Summary of Changes
 * add support for passive staking to wallet
 * clean up wallet copy pasta while there
 * multinode demo stakes validators with passive stake accounts

CC #4213 checkboxes 4 and 5

### CODE REVIEW

File: multinode-demo/fullnode.sh [hunk 1]
```diff
@@ -72,32 +72,51 @@ rsync_url() { # adds the 'rsync://` prefix to URLs that need it
...
   declare entrypoint_ip=$1
   declare node_keypair_path=$2
   declare vote_keypair_path=$3
-  declare stake=$4
+  declare stake_keypair_path=$4
+  declare stake=$5
 
-  declare node_keypair
-  node_keypair=$($solana_wallet --keypair "$node_keypair_path" address)
+  declare node_id
```
Comment (CriesofCarrots) [inline multinode-demo/fullnode.sh]: Didn't @mvines just change these to `_keypair` for precision? Why change back?
(maybe just a rebase thing?)

File: multinode-demo/fullnode.sh [hunk 2]
```diff
@@ -72,32 +72,51 @@ rsync_url() { # adds the 'rsync://` prefix to URLs that need it
...
+  node_id=$($solana_wallet --keypair "$node_keypair_path" address)
 
-  declare vote_keypair
-  vote_keypair=$($solana_wallet --keypair "$vote_keypair_path" address)
+  declare vote_id
+  vote_id=$($solana_wallet --keypair "$vote_keypair_path" address)
 
-  if [[ -f "$vote_keypair_path".configured ]]; then
-    echo "Vote account has already been configured"
+  declare stake_id
```
Comment (CriesofCarrots) [inline multinode-demo/fullnode.sh]: `stake_keypair`? (see above)

File: multinode-demo/fullnode.sh (hunk 1 above)
Comment (rob-solana) [reply multinode-demo/fullnode.sh]: it was a rebase thing, but....

it's really not correct to call what you get from $(wallet address) a keypair, it's only the pubkey..

File: multinode-demo/fullnode.sh (hunk 2 above)
Comment (rob-solana) [reply multinode-demo/fullnode.sh]: > r+, with one question in `fullnode.sh`
> Thanks for the `App` changes in wallet. `wallet.rs` is getting pretty unwieldy; this PR makes me consider several ways to reorganize, but no need to hold up your functionality

yeah: there were a bunch of deltas between the two builder stacks

**Comment by rob-solana** [comment]
> Thanks for the `App` changes in wallet. `wallet.rs` is getting pretty unwieldy; this PR makes me consider several ways to reorganize, but no need to hold up your functionality
it'd be nice if each of the commands were by itself with its test, decoder, and handler...

File: multinode-demo/fullnode.sh (hunk 1 above)
Comment (rob-solana) [reply multinode-demo/fullnode.sh]: updated with yet another way of naming it ;)

**Comment by rob-solana** [comment]
@mvines feels like dueling rebases

**Comment by mvines** [comment]
I'm done dorking around in multinode-demo/ for a while, have another bone to chew on now 🍖 
weak r+ from me, I haven't dug through this full PR but I assume @CriesofCarrots has :)

File: multinode-demo/fullnode.sh (hunk 1 above)
Comment (CriesofCarrots) [reply multinode-demo/fullnode.sh]: Good point! Should have looked closer. <3 pubkey

**Comment by CriesofCarrots** [comment]
> it'd be nice if each of the commands were by itself with its test, decoder, and handler...

I was thinking of sorting by native program (system, bpf_loader, budget, stake, vote), ie. domain of use, but individual commands is an interesting idea.

**Comment by rob-solana** [comment]
> > it'd be nice if each of the commands were by itself with its test, decoder, and handler...
> 
> I was thinking of sorting by native program (system, bpf_loader, budget, stake, vote), ie. domain of use, but individual commands is an interesting idea.

cool.  any grouping where the 4 or 5 places the command and parameters can be written and tested together would help a lot

**Comment by rob-solana** [comment]
mebbe wallet should have a per-program plugin API...

**Comment by mvines** [comment]
Or we just take the 🔪 to wallet and carve it up into a number of individual programs.  The voting setup stuff doesn't really sit right in a "wallet" to me but IMO we should let it get a little bit more bloated first to help inform us on where to make the cuts.

**Pull request merged**